src/object/cli_obj.c

```diff
@@ -4379,6 +4382,39 @@ obj_comp_cb(tse_task_t *task, void *data)
 			obj_ec_fail_info_reset(&obj_auxi->reasb_req);
 	}
 
+	if (unlikely(task->dt_result == -DER_TX_ID_REUSED)) {
+		D_ASSERT(daos_handle_is_inval(obj_auxi->th));
+
+		if (obj_auxi->retry_cnt != 0)
+			/* XXX: it is must because miss to set "RESEND" flag, that is bug. */
+			D_ASSERTF(0,
+				  "Miss 'RESEND' flag (%x) when resend the RPC for task %p: %u\n",
+				  obj_auxi->flags, task, obj_auxi->retry_cnt);
+
+		/* For non-retry case, restart TX with new TX ID. */
+		obj_auxi->io_retry = 1;
+		switch (obj_auxi->opc) {
+		case DAOS_OBJ_RPC_UPDATE: {
+			struct shard_rw_args	*rw_arg = &obj_auxi->rw_args;
+
+			D_INFO("DTX ID "DF_DTI" for update is reused, re-generate\n",
+			       DP_DTI(&rw_arg->dti));
+			daos_dti_gen(&rw_arg->dti, false);
+		}
+		case DAOS_OBJ_RPC_PUNCH:
+		case DAOS_OBJ_RPC_PUNCH_DKEYS:
+		case DAOS_OBJ_RPC_PUNCH_AKEYS: {
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment